packages/emotion-primitives/src/splitProps.js

```diff
@@ -0,0 +1,57 @@
+import {
+  textProps,
+  imageProps,
+  viewProps,
+  textStyleProps,
+  viewStyleProps,
+  imageStyleProps
+} from './props'
+
+function isValidStyleProp(element, propName) {
+  if (element === 'Text') {
+    return textStyleProps.includes(propName)
+  } else if (element === 'View') {
+    return viewStyleProps.includes(propName)
+  } else if (element === 'Image') {
+    return imageStyleProps.includes(propName)
+  }
+
+  return false
+}
+
+export function splitProps(rootEl, props) {
+  const rest = { ...props }
```

Comment:
this will make a copy of `props`, seems redundant as u only read from it anyway

Reply:
Haha ya I think I forgot to refactor before pushing the code. We can remove this since 'props' is pass by value.